### ISSUE TITLE
old haml dependency

### ISSUE THREAD

**Issue by mibamur** [k0kubun/hamlit#77]
Hello again!
There are some gem like gem 'opal-rails' which depend on HAML::Filters

worked only with `require: false`
```rb
# Gemfile

# gem 'haml'
gem 'haml-rails', require: false
gem 'hamlit', github: 'k0kubun/hamlit', submodules: true
gem 'hamlit-rails', github: 'mfung/hamlit-rails'
```

May you can tell me best solution how fix it.

**Comment by mibamur** [comment]
## example

### Other gem call often some thing like this

https://github.com/opal/opal-rails/blob/master/lib/opal/rails/haml_filter.rb#L1

```rb
module Haml::Filters::Opal
  include Haml::Filters::Base
...
```

**Comment by k0kubun** [comment]
Filters implemented with `Haml::Filters::Base` can't be used for Hamlit. You should re-implement it with `Hamlit::Filters::Base`.

I think we should create `Hamlit::Filters::Opal` in opal-rails/lib/opal/rails/hamlit_filter.rb. Since Hamlit's backend is the same as Slim, it'll be similar to `Slim::Embedded::OpalEngine`.

**Comment by mibamur** [comment]
Ok - try it now. Thanks

Here is another re-implement solution

Did you see it? faster_path https://github.com/danielpclark/faster_path/blob/master/lib/faster_path/optional/monkeypatches.rb

### I want something like this monkeypatches

**Comment by k0kubun** [comment]
Monkey-patching is basically a bad idea. It makes hard to find where the method is defined. In addition, monkey-patching implementation is fragile since it's separately maintained in spite of tight coupling.  Thus I'm reluctant to replace Haml's constant or methods in Hamlit.

Since Hamlit's goal is to be a replacement of Haml and haml.gem itself does not contain the filter for Opal, I don't accept adding Opal filter in hamlit.gem. So closing this issue.

**Comment by k0kubun** [comment]
I recommend two solutions to solve this issue.

- Adding `Hamlit::Filters::Opal` to opal-rails
- Creating something like hamlit-opal or opal-hamlit.

**Comment by k0kubun** [comment]
> Creating something like hamlit-opal or opal-hamlit

Here it is https://github.com/hamlit/hamlit-opal.
It's available as hamlit-opal v0.0.1.beta1. It's just conceptual and never tested.

**Comment by mibamur** [comment]
great
